datastreams/windower/tumbling.go

```diff
@@ -0,0 +1,104 @@
+package windower
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/elastiflow/pipelines/datastreams/internal/partition"
+	"github.com/elastiflow/pipelines/datastreams/internal/pipes"
+)
+
+type status struct {
+	started bool
+	mu      sync.Mutex
+}
+
+func newStatus() *status {
+	return &status{
+		started: false,
+	}
+}
+
+func (t *status) running() bool {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	return t.started
+}
+
+func (t *status) set(state bool) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	t.started = state
+}
+
+// tumbling buffers items for windowDuration once the first item arrives.
+// After windowDuration, it publishes the batch and clears it, then waits
+// for the next item to start a new timer.
+type tumbling[T any] struct {
+	*partition.Base[T]
```

Comment:
It probably makes more sense to embed this. 

```suggestion
	partition.Base[T]
```

Saves us an allocation and pointer hop while decreasing the GC footprint.